### PR TITLE
Glossary PoC

### PR DESCRIPTION
**Pull request by kodster28**
To dos:
- [x] Add related products to a products array & use that for filtering. Should also filter off productName instead of file name.
- [x] Render products array in the main table
- [x] Create glossary definition and glossary tooltip components based off [Kubernetes docs](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#glossary)
- [x] Adjust styling so glossary terms don't take up as much space.
- [x] Fix move definitions into glossary itself as opposed to separate .md files + replace current definition partials with glossary definition entries.

### CODE REVIEW

**Comment by kodster28** [comment]
Implements the following:
- [Top-level glossary](https://glossary-poc.cloudflare-docs-7ou.pages.dev/glossary/)
- [Product-level glossary](https://glossary-poc.cloudflare-docs-7ou.pages.dev/cloudflare-one/glossary/)
- [Term definitions](https://glossary-poc.cloudflare-docs-7ou.pages.dev/load-balancing/understand-basics/pools/) (the first bit of text)
- [Tooltip definitions](https://glossary-poc.cloudflare-docs-7ou.pages.dev/load-balancing/understand-basics/health-details/#components) (on both links and regular text).

On the tooltip definitions w/ links, there is a double tab focus to get to the link content.... but think that makes sense as it's providing two different options (hear a definition || visit the link).

File: layouts/shortcodes/glossary-definition.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{{- $id := .Get "term_id" -}}
+{{- $length := default "long" (.Get "length") -}}
+{{- $prepend := .Get "prepend" -}}
+
+{{- $glossaryDefinition := partial "glossary-definition.html" (dict "id" $id "length" $length "prepend" $prepend) -}}
```
Comment (kodster28) [inline layouts/shortcodes/glossary-definition.html:5]: Pulls from partial value that standardizes the extraction of the definition.

File: layouts/shortcodes/glossary-tooltip.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{- $id := .Get "term_id" -}}
+{{- $length := "short" -}}
+{{- $prepend := .Get "prepend" -}}
+{{- $text := .Inner -}}
+{{- $link := .Get "link" -}}
+
+{{- $glossaryDefinition := partial "glossary-definition.html" (dict "id" $id "length" $length "prepend" $prepend) -}}
```
Comment (kodster28) [inline layouts/shortcodes/glossary-tooltip.html:7]: Pulls from partial value that standardizes the extraction of the definition.

File: layouts/shortcodes/glossary.html [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+{{- range $k, $v := $glossaryData -}}
+{{- if in $v.products $product -}}
+{{- $glossaryEntries = merge $glossaryEntries (dict $k $v) -}}
+{{- end -}}
+{{- end -}}
+{{- else -}}
+{{- $glossaryEntries = $glossaryData -}}
+{{- end -}}
+
+<div class="selectorDropdowns">
```
Comment (kodster28) [inline layouts/shortcodes/glossary.html:15]: This is very similar to the `full-changelog` component

**Review by pedrosousa — APPROVED** [review]
A few partial usages need fixing in learning paths, but otherwise approved.

**Pull request merged**